experimental/oak_baremetal_runtime/src/framing.rs

```diff
@@ -25,69 +28,112 @@ use oak_remote_attestation::handshaker::{
 };
 use oak_remote_attestation_sessions::SessionId;
 
-struct InvocationHandler<F, G, V>
+enum AttestationState<G, V>
+where
+    G: AttestationGenerator,
+    V: AttestationVerifier,
+{
+    Uninitialized(AttestationBehavior<G, V>),
+    Initialized(Box<dyn AttestationTrait>),
+}
+
+struct InvocationHandler<G, V>
 where
-    F: Send + Sync + Clone + FnOnce(Vec<u8>) -> anyhow::Result<Vec<u8>>,
     G: AttestationGenerator,
     V: AttestationVerifier,
 {
-    attestation_handler: AttestationHandler<F, G, V>,
+    attestation_state: AttestationState<G, V>,
```

Comment:
I'm not sure this is the attestation state, shouldn't it be just initializationstate or just state? AFAICT the Wasm module is being initialized, not the attestation.

Reply:
Renamed to InitializationState, probably a better description. It's only `AttestationState` in so far as that it governs whether the attestation logic can call into the Wasm module